### PR TITLE
[add]add unit tests for events

### PR DESCRIPTION
**Pull request by fewdan**
### What problem does this PR solve?

add unit tests for events

### What is changed and how does it work?
add unit tests for events

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/apiserver/event/event_test.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("test err", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events?uid=err", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
```
Comment (WangXiangUSTC) [inline pkg/apiserver/event/event_test.go:236]: this line can be deleted

File: pkg/apiserver/event/event_test.go [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+					Duration:     "1m",
+					Pods:         nil,
+					ExperimentID: "testUID",
+				},
+			}
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=testKind", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
```
Comment (WangXiangUSTC) [inline pkg/apiserver/event/event_test.go:264]: ditto

File: pkg/apiserver/event/event_test.go [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/get?id=1", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusInternalServerError))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("other err", func() {
```
Comment (WangXiangUSTC) [inline pkg/apiserver/event/event_test.go:339]: what is the difference between "other err" and the "not found" above?

File: pkg/apiserver/event/event_test.go (hunk 2 above)
Comment (WangXiangUSTC) [reply pkg/apiserver/event/event_test.go:264]: and have many same problems in other case

File: pkg/apiserver/event/event_test.go (hunk 3 above)
Comment (fewdan) [reply pkg/apiserver/event/event_test.go:327]: In API processing, if the error is "Gorm recordnotfounderror", the returned results are different, so there are two cases.

File: pkg/apiserver/event/event_test.go [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+		}
+		router = gin.Default()
+		r := router.Group("/api")
+		endpoint := r.Group("/events")
+		endpoint.GET("", s.listEvents)
+		endpoint.GET("/dry", s.listDryEvents)
+		endpoint.GET("/get", s.getEvent)
+	})
+
+	AfterEach(func() {
```
Comment (g1eny0ung) [inline pkg/apiserver/event/event_test.go:188]: If there don't a need for this hook, I think it's better to remove it.

File: pkg/apiserver/event/event_test.go [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+			Expect(rr.Code).Should(Equal(http.StatusOK))
+			responseBody, err := json.Marshal(response)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("test err", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=err", nil)
+			Expect(err).ShouldNot(HaveOccurred())
```
Comment (g1eny0ung) [inline pkg/apiserver/event/event_test.go:270]: Shoud be `Expect(err).Should(HaveOccurred())`?

**Review by g1eny0ung — CHANGES_REQUESTED** [review]

File: pkg/apiserver/event/event_test.go (hunk 5 above)
Comment (fewdan) [reply pkg/apiserver/event/event_test.go:270]: This is the err returned by "NewRequest" , which should be nil.

File: pkg/apiserver/event/event_test.go (hunk 4 above)
Comment (fewdan) [reply pkg/apiserver/event/event_test.go:188]: image

In the other "*_ test.go" , the same method is used, so I moved here directly.

Comment (fewdan) [reply pkg/apiserver/event/event_test.go:188]: It should be "Add any teardown steps that needs to be executed after each test" here, let me change it.

Comment (g1eny0ung) [reply pkg/apiserver/event/event_test.go:188]: Ok. 👌🏽

File: pkg/apiserver/event/event_test.go (hunk 5 above)
Comment (g1eny0ung) [reply pkg/apiserver/event/event_test.go:270]: 🤔 Know about it. But it seems that all tested requests returned `nil`. Is it necessary to make this step of assertion?

Comment (fewdan) [reply pkg/apiserver/event/event_test.go:270]: OK, I'll take these out.

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Review by g1eny0ung — APPROVED** [review]
LGTM

**Pull request merged**